File: Fibonacci_Pattern/fib.go
package main

import (
    "fmt"
    "os"
    "strconv"
)

func fib(n int) {
    a, b := 0, 1
    for i := 0; i < n; i++ {
        a, b = b, a+b
        fmt.Printf("%d: %d\n", i+1, a)
    }
}

const msg = "Usage: please input the count of fibonacci numbers to output"

func main() {
    if len(os.Args) == 1 {
        fmt.Println(msg)
        return
    }
    if os.Args[1] == "" {
        fmt.Println(msg)
        return
    }
    n, err := strconv.Atoi(os.Args[1])
    if err != nil {
        fmt.Println(msg)
        return
    }
    fib(n)
}

File: Fibonacci_Pattern/fib.py
value: int = int(input())
    
def fib_by_recursion(data: int):
    if data <= 1:
        return data
    else:
        return fib_by_recursion(data - 1) + fib_by_recursion(data - 2)

def fibonacci(data: int):
    sequence: list = []
    index: int = 0
    while index < data + 1:
        if index == 0 or index == 1:
            sequence.append(index)
        else:
            sequence.append(sequence[index - 1] + sequence[index - 2])
        index += 1
    return sequence[-1]
